src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs

```diff
@@ -1,24 +1,160 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.IO;
-using System.Xml.Linq;
-
-#nullable enable
-
-namespace Microsoft.DotNet.XHarness.TestRunners.Xunit;
-
-internal class WasmXmlResultWriter
-{
-    public static void WriteOnSingleLine(XElement assembliesElement)
-    {
-        using var ms = new MemoryStream();
-        assembliesElement.Save(ms);
-        ms.TryGetBuffer(out var bytes);
-        var base64 = Convert.ToBase64String(bytes, Base64FormattingOptions.None);
-        Console.WriteLine($"STARTRESULTXML {bytes.Count} {base64} ENDRESULTXML");
-        Console.WriteLine($"Finished writing {bytes.Count} bytes of RESULTXML");
-    }
-}
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Xml.Linq;
+using System.Text;
+using System.Security.Cryptography;
+
+#nullable enable
+
+namespace Microsoft.DotNet.XHarness.TestRunners.Xunit;
+
+internal class WasmXmlResultWriter
+{
+#if DEBUG
+    public static void WriteOnSingleLine(XElement assembliesElement)
+    {
+        using var ms = new MemoryStream();
+        assembliesElement.Save(ms);
+        ms.TryGetBuffer(out var bytes);
+        var base64 = Convert.ToBase64String(bytes, Base64FormattingOptions.None);
```

Comment:
I think that base64 could contain `\n` in the middle right ? If so, you can insert some which would make the buffer in emscripten smaller. Because `stdout` emulator would build buffer until newline and only then write it into browser `console.log`.

 @maraf noted that maybe we had some issues with message order WRT `WASM EXIT` ... so I don't know if multi-line is good idea or not.

Reply:
We could in theory write each block as we transform it rather than create a single buffer. We should really review the emscripten line buffering.

Reply:
Alternatively, this could be `fetch('POST', xml)` in browser and nodejs. 
Remaining is `v8` shell which doesn't have file write. So we may need to keep this contraption.

Reply:
I really thing we should implement it as POST. Do we run V8 in CI at all ?

Reply:
> Do we run V8 in CI at all ?

yes, we noticed it in https://github.com/dotnet/runtime/pull/102069#issuecomment-2107048126. it'd be nice to fix that though.

Reply:
https://github.com/dotnet/xharness/pull/1388